builder/relinker/relinker.py

```
@@ -0,0 +1,463 @@
+                if index in l and t.file.split('.')[0] not in l:
+                    for m in self.targets:
+                        index = '*%s:(EXCLUDE_FILE'%(m.lib)
+                        if index in l and m.file.split('.')[0] not in l:
+                            l = l.replace('EXCLUDE_FILE(', 'EXCLUDE_FILE(%s '%(m.desc))
+                            if len(m.isecs) > 0:
+                                l += '\n    %s(%s)'%(m.desc, ' '.join(m.isecs))
+                    return l
+
+        return None
+
+    def save(self, input, output):
```

Ruff (0.15.7)

[error] Function argument `input` is shadowing a Python builtin

(A002)